frontend/src/views/File.vue

```diff
@@ -207,13 +221,27 @@ img.product
           </dl>
         </section>
       </section>
-      <section id="history">
+      <section id="provenance" v-if="response.sourceFileIds">
+        <header>Source files</header>
+        <section class="details">
+          <span class="notice">This file was generated using the following files:<br></span>
+            <div v-for="sourceFile in sourceFiles" :key="sourceFile.uuid" class="sourceFileList">
+              <router-link :to="`/file/${sourceFile.uuid}`">
+                <img :src="getIconUrl(sourceFile.product.id)" class="product">
+                {{ sourceFile.product.humanReadableName }}
+              </router-link><br>
+            </div>
+        </section>
+      </section>
+      <section id="history" v-if="response.history">
         <header>History</header>
-        <section class="details" v-if="response.history">
+        <section class="details" v-if="showHistory">
           <span class="monospace">{{ response.history.trim() }}</span>
-          <span class="notice">This is a non-standardized history provided by the file creator/processor.</span>
+          <span class="notice">This is a non-standardized history provided by the file creator/processor.
+          <a href="" @click.prevent="showHistory = false" id="hideHistory">Hide history</a>.</span>
         </section>
-        <section class="details na" v-else>N/A
+        <section class="details centered" v-else>
+          <a href="" @click.prevent="showHistory = true" id="showHistory">Show history</a>
         </section>
       </section>
```

Comment:
Harkitsitko, että piilottaisi koko laatikon eikä vain sen sisältöä? Jos tuo laatikko tulee jonkun ison kuvan viereen (en ole varma käykö niin käytännössä ikinä) niin siitä tulee iso tyhjä loota missä on tekstiä keskellä, mikä vie turhaa tilaa sivulta ja näyttää oudolta. Keskitetty teksti on ehkä muutenkin vähän erikoinen (ainakin jos se laatikko menee isoksi..), kun muualla sivulla ei ole sellaista?! Laittaisin itse show (ehkä "expand"?) / hide linkit joko otsikkopalkkiin (ja piilottaisin tyhjän lootan kokonaan) tai laittaisin sen samalla tavalla kuin "Preview not available" Preview lootassa. "Hide history" on myös nyt eri fonttikoolla kuin  "Show history" mikä on vähän häiritsevää. Olisiko sille joku parempi paikka, esim. omalla rivillä jos et halua laittaa sitä otsikkopalkkiin?